roles/change_hostname/files/change_hostname.rb

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/ruby
+
+# BEFORE RUNNING THIS SCRIPT
+#
+# Log into UI, rename the default capsule under infrastructure->capsules to the new hostname in both the name and url fields
+#
+#
+require "socket"
+
+STDOUT.puts "Starting the script run..."
+
+NEW_HOSTNAME = ARGV[0]
+
+# Get the hostname from your system
+OLD_HOSTNAME = Socket.gethostname
+
+# Check if the default NEW_HOSTNAME variable is updated in vars/main.yml
+if NEW_HOSTNAME == "changeme"
+  STDOUT.puts "Please update the script variables with your new hostname"
+  exit
+end
+
+STDOUT.puts "stopping services"
+`katello-service stop`
+
+STDOUT.puts "deleting old certs"
+`
+rm -rfv /etc/pki/katello{,.bak}
+rm -rfv /etc/pki/katello-certs-tools{,.bak}
+rm -rfv /etc/candlepin/certs/amqp{,.bak}
+rm -rfv /etc/foreman-proxy/*ssl*
+rm -rfv /etc/foreman/old-certs
+rm -rfv /etc/foreman/*.pem
+rm -rfv /var/lib/puppet/ssl
+rm -rfv /root/ssl-build
+`
+
+STDOUT.puts "updating hostname in /etc/hostname"
+`sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/hostname`
+
+STDOUT.puts "updating hostname in /etc/hosts"
+`sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/hosts`
+
+STDOUT.puts "setting hostname"
+`hostnamectl set-hostname #{NEW_HOSTNAME}`
```

Comment:
@pgagne is right, we still have to change this - @sthirugn I suggest we check for version and then update the hostname differently if its rhel6 or 7, I can write something when I have a minute